### PR TITLE
Split SwapForm

### PR DESCRIPTION
**Pull request by hungdoansy**
# Description
I split the SwapForm component from pages/SwapV2/index.tsx so that later on we can easily replace this SwapForm with a new one (which uses new Meta Aggregator APIs).

The component is placed in components/swapv2/SwapForm.tsx

In this PR, I did:
- Just move the SwapForm into a separate file. Some hooks are being used twice
- Do refactoring on Routing component (remove `trade` from props)
- Remove `useDerivedSwapInfoV2` and use other hooks instead. `useDerivedSwapInfoV2` should be used only in SwapFormV2 component

### CODE REVIEW

File: src/state/swap/hooks/index.ts [hunk 1]
```diff
@@ -430,3 +430,14 @@ export const useDefaultsFromURLSearch = ():
...
 }
+
+export const useInputCurrency = () => {
+  const inputCurrencyId = useSelector((state: AppState) => state.swap[Field.INPUT].currencyId)
+  const inputCurrency = useCurrencyV2(inputCurrencyId)
+  return inputCurrency || undefined
+}
+export const useOutputCurrency = () => {
+  const outputCurrencyId = useSelector((state: AppState) => state.swap[Field.OUTPUT].currencyId)
+  const outputCurrency = useCurrencyV2(outputCurrencyId)
```
Comment (nguyenhoaidanh) [inline src/state/swap/hooks/index.ts:441]: useDerivedSwapInfoV2 use useCurrency
maybe conflict.

File: src/state/swap/hooks/useParsedAmountFromInputCurrency.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import { tryParseAmount } from '.'
+
+// from the current swap inputs, compute the best trade and return it.
+const useParsedAmountFromInputCurrency = () => {
+  const typedValue = useSelector((state: AppState) => state.swap.typedValue)
+  const inputCurrency = useInputCurrency()
+  const outputCurrency = useOutputCurrency()
+
+  const isExactIn = true
```
Comment (nguyenhoaidanh) [inline src/state/swap/hooks/useParsedAmountFromInputCurrency.ts:15]: double check

File: src/state/swap/hooks/useParsedAmountFromTypedInput.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react'
+import { useSelector } from 'react-redux'
```
Comment (viet-nv) [inline src/state/swap/hooks/useParsedAmountFromTypedInput.ts:2]: useAppSelector maybe better

Comment (viet-nv) [reply src/state/swap/hooks/useParsedAmountFromTypedInput.ts:2]: https://redux-toolkit.js.org/tutorials/typescript#define-typed-hooks

**Review by viet-nv — DISMISSED** [review]

**Pull request closed without merging**